### ISSUE TITLE
xformers `Version: 0.0.26.post1` is clashing with torch 2.3.1

### ISSUE THREAD

**Issue by MoezGholami** [facebookresearch/xformers#1052]
# 🐛 Bug

installing xformers `pip install xformers` (which at the moment points to `0.0.26.post1` uninstalls torch and triton 2.3.1 and replaces it with 2.3.0. This is causing conflict with the packages that do require 2.3.1 such as "torchvision 0.18.1+cu121" and "torchaudio 2.3.1+cu121".

## To Reproduce

```
$ pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu121
[skipped output]
$ pip install xformers==0.0.26.post1 # (currently equivalent to pip install xformers)
[...]
Collecting xformers
  Using cached xformers-0.0.26.post1-cp310-cp310-manylinux2014_x86_64.whl (222.7 MB)
Collecting torch==2.3.0
  Using cached torch-2.3.0-cp310-cp310-manylinux1_x86_64.whl (779.1 MB)
[...]
  Attempting uninstall: triton
    Found existing installation: triton 2.3.1
    Uninstalling triton-2.3.1:
      Successfully uninstalled triton-2.3.1
      [...]
  Attempting uninstall: torch
    Found existing installation: torch 2.3.1+cu121
    Uninstalling torch-2.3.1+cu121:
[...]
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
torchvision 0.18.1+cu121 requires torch==2.3.1, but you have torch 2.3.0 which is incompatible.
torchaudio 2.3.1+cu121 requires torch==2.3.1, but you have torch 2.3.0 which is incompatible.

```

## Expected behavior

xformers working with latest version of torch.

**Comment by ashleykleynhans** [comment]
I also need xformers to be compatible with torch 2.3.1 but I would say that this is a feature request rather than a bug.

**Comment by comaniac** [comment]
Second on this issue.

**Comment by dmaljovec** [comment]
This is not as simple as changing [this line](https://github.com/facebookresearch/xformers/blob/133d7f1cb0a050f9ee8b8cd02fd9c906247e6c4e/.github/workflows/wheels.yml#L38) to:

```
            for torch_version in ['2.3.0', '2.3.1']:
```

Is it? If I understand the tests correctly, my guess is this is already tested with a later version given the use of the nightly channel from Pytorch used [here](https://github.com/facebookresearch/xformers/blob/133d7f1cb0a050f9ee8b8cd02fd9c906247e6c4e/.github/workflows/gpu_test_gh.yml#L32-L35).

**Comment by dmaljovec** [comment]
FWIW, I am experimenting with using the ["install from source"](https://github.com/facebookresearch/xformers#:~:text=pre%20%2DU%20xformers-,Install%20from%20source,-%3A%20If%20you%20want) directions and it is at least able to resolve an environment. I have yet to test that everything works together, but I am at least able to generate a lockfile with [uv](https://github.com/astral-sh/uv) combining `xformers@git+https://github.com/facebookresearch/xformers.git@main#egg=xformers` and `torch>=2.3.1`.

**Comment by comtest99** [comment]
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
xformers 0.0.26.post1 requires torch==2.3.0, but you have torch 2.3.1+cu121 which is incompatible.

**Comment by danthe3rd** [comment]
Hi,
We don't have a release for 2.3.1, we will make a release for 2.3.2 tho or 2.4.0. In the meantime, you can either downgrade pytorch, or install xFormers from source (long).

**Comment by danthe3rd** [comment]
Good point - in that case we will try to add a release for 2.3.1 then

**Comment by chi0tzp** [comment]
> Good point - in that case we will try to add a release for 2.3.1 then

Thanks @danthe3rd, much appreciated! Any ETA for the 2.3.1 build? Cheers!

**Comment by danthe3rd** [comment]
We'll try to get dev wheels/conda packages early next week, and hopefully soon after that some stables for 2.3.1

**Comment by chi0tzp** [comment]
Cheers @danthe3rd!

**Comment by danthe3rd** [comment]
Version 0.0.27 just got published, and is compatible with PT 2.3.1
